dht_bootstrap.go

```diff
@@ -117,60 +95,42 @@ func (dht *IpfsDHT) bootstrapBuckets(ctx context.Context) error {
 	}
 
 	buckets := dht.routingTable.GetAllBuckets()
-	var wg sync.WaitGroup
-	errChan := make(chan error)
-
+	if len(buckets) > 16 {
+		// Don't bother bootstrapping more than 16 buckets.
+		// GenRandPeerID can't generate target peer IDs with more than
+		// 16 bits specified anyways.
+		buckets = buckets[:16]
+	}
 	for bucketID, bucket := range buckets {
-		if time.Since(bucket.RefreshedAt()) > dht.bootstrapCfg.BucketPeriod {
-			wg.Add(1)
-			go func(bucketID int, errChan chan<- error) {
-				defer wg.Done()
-				// gen rand peer in the bucket
-				randPeerInBucket := dht.routingTable.GenRandPeerID(bucketID)
-
-				// walk to the generated peer
-				walkFnc := func(c context.Context) error {
-					_, err := dht.FindPeer(ctx, randPeerInBucket)
-					if err == routing.ErrNotFound {
-						return nil
-					}
-					return err
+		if time.Since(bucket.RefreshedAt()) > dht.bootstrapPeriod {
+			// gen rand peer in the bucket
+			randPeerInBucket := dht.routingTable.GenRandPeerID(bucketID)
+
+			// walk to the generated peer
+			walkFnc := func(c context.Context) error {
+				_, err := dht.FindPeer(ctx, randPeerInBucket)
```

Comment:
Should be using the passed-in context, i.e. `c`?

Reply:
Yes.